File: library/src/main/java/nl/endevelopment/r2randroid/r2rlib/models/SurfaceLineCode.java
package nl.endevelopment.r2randroid.r2rlib.models;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by jan on 11/07/16.
 */

public class SurfaceLineCode implements Parcelable {
    private String code; // 	Internal line code

    public SurfaceLineCode() {
    }

    public SurfaceLineCode(String code) {
        this.code = code;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }



    protected SurfaceLineCode(Parcel in) {
        code = in.readString();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(code);
    }

    @SuppressWarnings("unused")
    public static final Parcelable.Creator<SurfaceLineCode> CREATOR = new Parcelable.Creator<SurfaceLineCode>() {
        @Override
        public SurfaceLineCode createFromParcel(Parcel in) {
            return new SurfaceLineCode(in);
        }

        @Override
        public SurfaceLineCode[] newArray(int size) {
            return new SurfaceLineCode[size];
        }
    };
}

File: library/src/main/java/com/getroadmap/r2rlib/models/SurfaceLineCode.kt
package com.getroadmap.r2rlib.models

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by jan on 11/07/16.
 */

class SurfaceLineCode() : Parcelable {
    var code: String? = null // 	Internal line code

    constructor(parcel: Parcel) : this() {
        code = parcel.readString()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(code)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<SurfaceLineCode> {
        override fun createFromParcel(parcel: Parcel): SurfaceLineCode {
            return SurfaceLineCode(parcel)
        }

        override fun newArray(size: Int): Array<SurfaceLineCode?> {
            return arrayOfNulls(size)
        }
    }

}